storage/range_command.go

```diff
@@ -864,6 +864,17 @@ func (r *Range) AdminSplit(args *proto.AdminSplitRequest, reply *proto.AdminSpli
 		return txn.Flush()
 	}); err != nil {
 		reply.SetGoError(util.Errorf("split at key %s failed: %s", splitKey, err))
+		return
+	}
+
+	if err := r.rm.ProcessRangeDescriptorUpdate(r.Desc().RaftID); err != nil {
```

Comment:
These need to be in the triggers, so the updated descriptor is available. In fact, we could probably just call ProcessRangeDescriptorUpdate in Range.SetDesc.